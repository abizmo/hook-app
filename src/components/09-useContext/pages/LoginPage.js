import React, { useContext } from 'react'
import UserContext from '../context/UserContext'

const LoginPage = () => {
  const { user, setUser } = useContext( UserContext )

  const handleLogin = () => {
    setUser({
      id: 123,
      name: 'abizmo',
      email: 'abizmo@abizmo.dev',
    })
  }

  return (
    <div>
      <h1>Login</h1>
      <hr />
      <button
        className="btn btn-primary"
        disabled={!!user.id}
        onClick={handleLogin}
      >
        Login
      </button>
    </div>
  )
}

export default LoginPage
