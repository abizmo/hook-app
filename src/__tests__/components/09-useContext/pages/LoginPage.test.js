import {mount} from 'enzyme'
import UserContext from "../../../../components/09-useContext/context/UserContext"
import LoginPage from "../../../../components/09-useContext/pages/LoginPage"

describe('LoginPage tests', () => {
  const user = {}
  const setUser = jest.fn()
  const wrapper = mount(
    <UserContext.Provider value={{user, setUser}}>
      <LoginPage />
    </UserContext.Provider>
  )

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  test('should call setUser', () => {
    wrapper.find('button').simulate('click')
    expect(wrapper.find('button').prop('disabled')).toBeFalsy()
    expect(setUser).toHaveBeenCalled()
  })
})
