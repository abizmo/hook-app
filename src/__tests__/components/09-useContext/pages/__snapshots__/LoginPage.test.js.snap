// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage tests should render and match snapshot 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
